### PR TITLE
Ft authentication endpoints

### LINKED ISSUE #5: Authentication backend
**Feature Description**
A user should be able to create an account on the platform and login.

**Tasks**
- [x] Create a [app.properties](https://github.com/BuildForSDG/.github/blob/master/app.properties) file in root of repo and fill details
- [x] Setup Flask env
- [x] Install PyLint
- [x] Build a register endpoint
- [x] Valite register input data
- [x] Build a login endpoint
- [x] Validate login input data
- [x] Authenticate with JWT
- [ ] Build reset password endpoint
- [ ] Wire up email verification

**Additional Context**
branch `ft-authentication-endpoints`

### PR DESCRIPTION
**Pull request by moseti1**

### CODE REVIEW

**Comment by moseti1** [comment]
installed pylint 
and tried building authentication endpoints

**Review by Dave-mash — COMMENTED** [review]
Great work Moseti. Collaborate with Jordan to come up with uniform endpoints

**Review by Dave-mash — DISMISSED** [review]

File: app.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+        next = flask.request.args.get('next')
+        # is_safe_url should check if the url is safe for redirects.
+        # See http://flask.pocoo.org/snippets/62/ for an example.
+        if not is_safe_url(next):
+            return flask.abort(400)
+
+        return flask.redirect(next or flask.url_for('index'))
+    return flask.render_template('login.html', form=form)
+
```
Comment (Dave-mash) [inline app.py:33]: Since were building an API return JSON data rather than a form template.

File: app.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        flask.flash('Logged in successfully.')
+
+        next = flask.request.args.get('next')
+        # is_safe_url should check if the url is safe for redirects.
+        # See http://flask.pocoo.org/snippets/62/ for an example.
+        if not is_safe_url(next):
+            return flask.abort(400)
+
+        return flask.redirect(next or flask.url_for('index'))
```
Comment (Dave-mash) [inline app.py:31]: This line is redundant.

File: backend/__init__.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from flask_sqlalchemy import SQLAlchemy
+
+# init SQLAlchemy so we can use it later in our models
+db = SQLAlchemy()
+@app.route("/")
+def create_app():
+    app = Flask(__name__)
+
+    app.config['SECRET_KEY'] = 'secret-key-goes-here'
+    app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///db.sqlite'
```
Comment (Dave-mash) [inline backend/__init__.py]: Merge work with Jordan he has setup a database connection with sqlalchemy

Comment (Dave-mash) [reply backend/__init__.py]: Also check out this repo for more ideas https://github.com/Dave-mash/Questioner-api-v2

File: app.py (hunk 1 above)
Comment (Jordan-type) [reply app.py:33]: change this to return JSON data using flask jsonify also to return error code responses

File: backend/__init__.py (hunk 3 above)
Comment (Jordan-type) [reply backend/__init__.py]: use postgresql database and create a config.py file with  development, production, and testing methods

# Production database
class Production(Config):
    SQLALCHEMY_DATABASE_URI = ''
    SQLALCHEMY_TRACK_MODIFICATIONS = False
    ENV = "production"
    DEBUG = False

**Review by Jordan-type — CHANGES_REQUESTED** [review]
Great work on the file structure check out developing flask Apis

**Pull request closed without merging**